versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5804 @@
+
+|var   |value|
+|------|-----|
+|test.a|["1", "2", "3"]|
+|test.b|["x"]|
+
+The task will run successfully, because `test.c` is not required. Given these values, the command would be instantiated as:
+
+```txt
+/bin/mycmd 1 2 3
+/bin/mycmd x
+/bin/mycmd
+```
+
+If the inputs were:
+
+|var   |value|
+|------|-----|
+|test.a|["1", "2", "3"]|
+|test.b|["x","y"]|
+|test.c|["a","b","c","d"]|
+
+Then the command would be instantiated as:
+
+```txt
+/bin/mycmd 1 2 3
+/bin/mycmd x,y
+/bin/mycmd /path/to/c.txt
+```
+
+##### Optional inputs with defaults
+
+It *is* possible to provide a default to an optional input type. This may be desirable in the case where you want to have a defined value by default, but you want the caller to be able to override the default and set the value to undefined (i.e. `None`).
+
+```wdl
+task say_hello {
+  input {
+    String name
+    String? saluation = "hello"
+  }
+  command {}
+  output {
+    String greeting = if defined(saluation) then "${saluation} ${name}" else name
+  }
+}
+
+workflow foo {
+  input {
+    String name
+    Boolean use_salutation
+  }
+  
+  if (use_salutation) {
+    call say_hello { input: name = name }
+  }
+  if (!use_salutation) {
+    call say_hello { 
+      input: 
+        name = name,
+        salutation = None 
+    }
+  }
+}
+```
+
+### Private Declarations
+
+A task can have declarations that are intended as intermediate values rather than inputs. These private declarations may appear anywhere in the body of the task, and they must be initialized. Just like input declarations, private declarations may be initialized with literal values, or with expressions that may reference other declarations.
+
+For example, this task takes an input and then performs a calculation, using a private declaration, that can then be referenced in the command template:
+
+```wdl
+task t {
+  input {
+    Int size
+  }
+
+  Int size_clamped = if size > 10 then 10 else size
+  ...
+}
+```
+
+The value of a private declaration may *not* be specified by the task caller, nor is it accessible outside of the current scope.
+
+```wdl
+task test {
+  input {
+    Int i
+  }
+  String s = "hello"
+  command { ... }
+  output {
+    File out = "/path/to/file"
+  }
+}
+
+workflow wf {
+  call test {
+    input:
+      i = 1,         # this is fine - "i" is in the input section
+      s = "goodbye"  # error! "s" is private
+  }
+  output {
+    File out = test.out # this is fine - "out" is in the output section
+    String s = test.s   # error! "s" is private
+  }
+}
+```
+
+### Command Section
+
+The `command` section is the only required task section. It defines the command template that is evaluated and executed when the task is called. Specifically, the commands are executed after all of the inputs are staged and before the outputs are evaluated.
+
+There are two different syntaxes that can be used to define the command section:
+
+```wdl
+# HEREDOC style - this way is preferred
```

Comment:
Oh, I stand corrected.  Then I would go back through and make sure all example command sections have HEREDOC styling with this exception here.

Reply:
done